File: RoomProject/DataBinding/app/src/main/java/com/shuworld/databinding/MainActivity.java
package com.shuworld.databinding;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.shuworld.databinding.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        fetchUserProfile();
    }

    private void fetchUserProfile() {
        UserProfile userProfile = new UserProfile();
        userProfile.name = "Hong Gil Dong";
        userProfile.phone = "010-1000-1000";
        userProfile.address = "수원시 권선동";

        binding.setUserProfile(userProfile);
        //updateUI(userProfile);
    }

/*
    private void updateUI(UserProfile userProfile) {

        // 데이터 바인딩

        binding.name.setText(userProfile.name);
        binding.phone.setText(userProfile.phone);
        binding.address.setText(userProfile.address);
    }
*/
}

File: Practice/DataBinding_Kotlin2/app/src/main/java/com/example/databinding_kotlin/MainActivity.kt
package com.example.databinding_kotlin

import android.database.DatabaseUtils
import android.databinding.DataBindingUtil.setContentView
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.example.databinding_kotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val DataBinding = DatabaseUtils.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        binding.vm = ViewModel()
    }
}